src/Bicep.Core/Configuration/ExperimentalFeaturesEnabled.cs

```diff
@@ -19,7 +19,8 @@ public record ExperimentalFeaturesEnabled(
     bool OptionalModuleNames,
     bool LocalDeploy,
     bool ResourceDerivedTypes,
-    bool SecureOutputs)
+    bool SecureOutputs,
+    bool DeploymentFile)
```

Comment:
FYI this PR should give you a good overview of the boilerplate needed for experimental features: #12212